mobile/native/src/max_quantity.rs

```diff
@@ -0,0 +1,304 @@
+use crate::calculations;
+use crate::channel_trade_constraints::channel_trade_constraints;
+use crate::ln_dlc;
+use bitcoin::Amount;
+use commons::order_matching_fee_taker;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal::Decimal;
+
+pub fn max_quantity(price: Decimal, trader_leverage: f32) -> anyhow::Result<Decimal> {
+    let channel_trade_constraints = channel_trade_constraints()?;
+
+    let on_chain_fee_estimate = match channel_trade_constraints.is_channel_balance {
+        true => None,
+        false => {
+            let channel_fee_reserve = ln_dlc::estimated_fee_reserve()?;
+            let funding_tx_fee = ln_dlc::estimated_funding_tx_fee()?;
+            // double the funding tx fee to ensure we have enough buffer
+            let funding_tx_with_buffer = funding_tx_fee * 2;
+
+            Some(channel_fee_reserve + funding_tx_with_buffer)
+        }
+    };
+
+    let max_coordinator_margin =
+        Amount::from_sat(channel_trade_constraints.max_counterparty_margin_sats);
+    let max_trader_margin = Amount::from_sat(channel_trade_constraints.max_local_margin_sats);
+
+    let max_quantity = calculate_max_quantity(
+        price,
+        max_coordinator_margin,
+        max_trader_margin,
+        on_chain_fee_estimate,
+        channel_trade_constraints.coordinator_leverage,
+        trader_leverage,
+    );
+
+    Ok(max_quantity)
+}
+
+fn calculate_max_quantity(
+    price: Decimal,
+    max_coordinator_margin: Amount,
+    max_trader_margin: Amount,
+    on_chain_fee_estimate: Option<Amount>,
+    coordinator_leverage: f32,
```

Comment:
eventually we should get rid of f32 and use Decimal in as many places as possible.